utils/scale-test

```diff
@@ -102,6 +102,8 @@ def run_once_with_primary(args, ast, rng, primary_idx):
         d = ensure_tmpdir(args.tmpdir)
         inputs = [write_input_file(args, ast, d, i) for i in rng]
         primary = inputs[primary_idx]
+        # frontend no longer accepts duplicate inputs
+        del inputs[primary_idx]
```

Comment:
This seems like we should just change the tests again. @graydon?

Reply:
I'm not sure I get what's going on here; isn't -primary-file just _identifying_ which of the input files is the primary? I.e. isn't it supposed to be providing the primary in the list-of-inputs as well? or is this PR all about changing that so that the list of inputs excludes the primary-files?

Reply:
It is a bit odd, but the way it works and worked before I messed with the code depends on whether the input files are on the command line or in a filelist. You can't have both.
If in a filelist, the primaries MUST be repeated in the filelist because order matters.
If on the command line, the primaries should not be repeated on the command line, because the -primary hooha argument(s) specify the order.
It used to be that we got duplicate filenames in the list of filenames for -primary-file foo.swift foo.swift on the command line. With my changes, a given filename must only appear once on the command line.

Reply:
@jordan I don't think there's anything to change in the tests; this function is responsible for composing synthetic command lines to repeatedly run instances of swiftc with. I think @davidungar is just adapting those synthetic command lines to the new requirements the PR is imposing (which is fine by me!) unless I misunderstand.